lightning/src/util/events.rs

```diff
@@ -135,6 +135,21 @@ pub enum Event {
 		/// The outputs which you should store as spendable by you.
 		outputs: Vec<SpendableOutputDescriptor>,
 	},
+	/// This event is generated when a payment has been successfully forwarded through us and a
+	/// forwarding fee earned.
+	PaymentForwarded {
+		/// The fee, in milli-satoshis, which was earned as a result of the payment.
+		///
+		/// Note that if we force-closed a channel while a forwarded HTLC was pending, the amount
+		/// the next hop claimed may have been rounded down, causing a higher fee than expected.
+		///
+		/// If the channel which sent us the payment has been force-closed, we will claim the funds
+		/// via an on-chain transaction. In that case we do not consider the on-chain transaction
+		/// fees involved in such a claim and this value will be `None`.
```

Comment:
Wait I think a `onchain_value_satoshis` value is passed in `process_pending_monitor_events` and consumed as part of the generated `PaymentForwarded` there ? Or are you thinking about another case ?

Reply:
Wait, I'm confused, those are only about claims that our counterparty did on-chain, causing us to update off-chain state. Did I miss something about how `HTLCUpdate` events are generated?

Reply:
I think Antoine's right, seems like this will always be non-`None` with the current code?

Reply:
If `claim_funds_from_hop` hits the `short_to_id.get(..) == None` branch right at the top, it'll return `(None, Err(None))` which gets translated into our `ChannelMonitorUpdate`-generation branch, but should still consider `claimed_htlc` to be true, resulting in `claimed_value_msat` being `None`.

Reply:
See updated doc commit, is that sufficient?